jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalityInspector.java

```diff
@@ -0,0 +1,232 @@
+ * by default. When {@link IterationOrder#LEX_BFS} is specified as a second constructor parameter, this
+ * {@code ChordalityInspector} uses {@link LexBreadthFirstIterator} to compute perfect elimination order.
+ * <p>
+ * A <a href="https://en.wikipedia.org/wiki/Chordal_graph">chordal graph</a> is one in which all cycles of
+ * four or more vertices have a chord, which is an edge that is not part of the cycle but connects two vertices
+ * of the cycle.
+ * <p>
+ * A graph is chordal iff its the vertices can be arranged into a perfect elimination order. More than one perfect
+ * elimination order may exist for a given graph. Either maximum cardinality search or lexicographical breadth-first
+ * search can be used to produce such an order.
+ * <p>
+ * Both lexicographical BFS and maximum cardinality search run in $\mathcal{O}(|V| + |E|)$. Checking whether given order
+ * is the perfect elimination order via {@link ChordalityInspector#isPerfectEliminationOrder(List)} takes
+ * $\mathcal{O}(|V| + |E|)$ as well. So, overall time complexity of the method
+ * {@link ChordalityInspector#isChordal()} is $\mathcal{O}(|V| + |E|)$.
+ *
+ * @param <V> the graph vertex type.
+ * @param <E> the graph edge type.
+ * @author Timofey Chudakov
+ * @see LexBreadthFirstIterator
+ * @see MaximumCardinalityIterator
+ * @since March 2018
+ */
+public class ChordalityInspector<V, E> {
+    /**
+     * The inspected graph.
+     */
+    private Graph<V, E> graph;
+    /**
+     * Contains true if the graph is chordal, otherwise false. Is null before the first call to the
+     * {@link ChordalityInspector#isChordal()}.
+     */
+    private Boolean chordal = null;
+    /**
+     * Order produced by {@code orderIterator}.
+     */
+    private List<V> order;
+    /**
+     * Iterator used for producing perfect elimination order.
+     */
+    private GraphIterator<V, E> orderIterator;
+
+    /**
+     * Creates a chordality inspector for {@code graph}, which uses {@link MaximumCardinalityIterator}
+     * as a default iterator.
+     *
+     * @param graph the graph for which a chordality inspector to be created.
+     */
+    public ChordalityInspector(Graph<V, E> graph) {
+        this(graph, IterationOrder.MCS);
+    }
+
+    /**
+     * Creates a chordality inspector for {@code graph}, which uses an iterator defined by the second
+     * parameter as an internal iterator.
+     *
+     * @param graph          the graph for which a chordality inspector is to be created.
+     * @param iterationOrder the constant, which defines iterator to be used by this {@code ChordalityInspector}.
+     */
+    public ChordalityInspector(Graph<V, E> graph, IterationOrder iterationOrder) {
+        this.graph = Objects.requireNonNull(graph);
+        if (graph.getType().isDirected()) {
+            this.graph = new AsUndirectedGraph<>(graph);
+        }
+        if (iterationOrder == IterationOrder.MCS) {
```

Comment:
You should store the iteration order in a private variable and use it instead of doing instanceof.